README.md

```diff
@@ -20,11 +26,148 @@ Or install it yourself as:
 
 ## Usage
 
-TODO: Write usage instructions here
+`ActiveService` defines the concept of `operation`. An operation is a method that will run some business function, for example creating a user. ActiveServices don't have state.
+
+### operation
+
+An example UsersService could look like this:
+
+```
+class UsersService
+	include ActiveService
+
+	# Ruby 2.1 required parameters :)
+	operation :create_user do |name:, email:|
+		user = User.create!(name: name, email: email)
+
+		UserMailer.welcome(user).deliver_later
+
+		user
+	end
+
+end
+```
+
+You can use this service like this:
+
+```
+UsersService.create_user(name: 'Jhon', email: 'jhon@x.com')
+```
+
+### pipe
+
+This service can be refactored to something like this:
+
+```
+class UsersService
+	include ActiveService
+
+	# `pipe` name will probably change.
+	pipe :create_user, :send_welcome_email
+	operation :create_user do |name:, email:|
+		User.create!(name: name, email: email)
+	end
+
+	private
+
+	def send_welcome_email(user)
+		UserMailer.welcome(user).deliver_later
+	end
+
+end
+```
+
+`pipe` allows you to process and compose the output of a service operation. You can define as many pipes you want for an operation, this will allow you to process the output like bash `|` does.
+
+### before / around / after
+
+You can use filters like ActionController.
+
+```
+class UsersService
+	include ActiveService
+
+	before :activate_account, :ensure_elegible
```

Comment:
I perfectly understand how the `before` works, but when i read it i got a bit confused.
which method comes before what?

I sujest something more descriptive like: `before :activate_account, run: :ensure_elegible`